src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml

```diff
@@ -51,4 +51,35 @@
       <attribute internal="RemoveAttributeInstances" />
     </type>
   </assembly>
+
+  <!-- COM specific attributes -->
+  <assembly fullname="System.Private.CoreLib" feature="System.Runtime.InteropServices.Marshal.IsComSupported" featurevalue="false">
+    <type fullname="System.MTAThreadAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.STAThreadAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.ClassInterfaceAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.CoClassAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.ComDefaultInterfaceAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.ComImportAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.ComVisibleAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Runtime.InteropServices.GuidAttribute">
```

Comment:
If we want to strip this one, we should make sure `System.Type::GUID` is somehow unusable. I've seen people use things like this creatively (e.g. I've seen people using the assembly's public key as a public key for encrypting communication, etc.). It wouldn't surprise me if people use this as a "type-associated unique ID". Stripping this attribute just silently changes the GUID.

Reply:
Yeah, that should be doable

Reply:
I was looking at the implementation and it returns empty GUID in this case. I guess that's better than to throw.

Reply:
Is that what Mono returns when there's no GuidAttribute? The runtime is supposed to compute a GUID using some weird algorithm from name, I think (it does so on CoreCLR).

Reply:
Btw the docs for [Type.GUID](https://docs.microsoft.com/en-us/dotnet/api/system.type.guid?view=netcore-3.1) don't mention COM anywhere so this might come as a surprise to some users.

Reply:
Yeah, Mono return empty GUID. Checking the CoreCLR implementation https://github.com/dotnet/runtime/blob/0a69f0262ee762252d2636b49b82c6030c386046/src/coreclr/src/vm/reflectioninvocation.cpp#L2148-L2171 it looks like this should work fine. Basically making no-com on Windows to behave the same as CoreCLR on Unix and generating some random GUID.

Reply:
In that case we need to make it so that the behavior is the same without linker in the picture. So this change would need to incorporate change to CoreCLR which looks at the runtime property and ignores the attribute in that case.

Reply:
I would leave the `GuidAttribute`. This is not a COM support thing. I have observe many projects use this for making their type's Guid X instead of it being generated. I think removing this is a bit aggressive.

Reply:
Thinking about this more, I would be fine removing the `GuidAttribute` but then I would argue for making `Type.Guid` match the behavior of mono.

My reasoning here is that mono is the canonical runtime implementation with no COM support. This means it should be considered the source of truth semantically speaking. Making this change would also help to validate the assumption that the `GuidAttribute` is only for COM scenarios in the sense that if we start returning `Guid.Empty` when one disables COM support and no one balks then the assumption is true if there are no other uses of COM in the system. However, if users complain that the behavior of `Type.Guid` has now changed in CoreCLR and we confirm they are not using COM in any manner than it invalidates that assumption and indicates that `Type.Guid` and therefore `GuidAttribute` are relied upon in non-COM scenarios.